src/components/tables/TableConnected.tsx

```diff
@@ -22,40 +22,48 @@ import { ITableHeaderCellState } from './TableHeaderCellReducers';
 import { ITableCompositeState, ITableState } from './TableReducers';
 import { getTableChildComponentId } from './TableUtils';
 
-const mapStateToProps = (state: IReactVaporState, ownProps: ITableOwnProps): ITableCompositeStateProps => {
-  const tableState: ITableState = state.tables[ownProps.id] || {} as ITableState;
-  const filterState: IFilterState = tableState && _.findWhere(state.filters, { id: tableState.filterId });
-  const paginationState: IPaginationState = tableState && _.findWhere(state.paginationComposite, { id: tableState.paginationId });
-  const perPageState: IPerPageState = tableState && _.findWhere(state.perPageComposite, { id: tableState.perPageId });
+export interface IReactVaporStateExtended extends IReactVaporState {
+  [additionalState: string]: any;
+}
+
+export const getTableCompositeState = (state: IReactVaporStateExtended, id: string): ITableCompositeState => {
+  const tableState: ITableState = state.tables[id] || {} as ITableState;
+  const filterState: IFilterState = tableState && _.findWhere(state.filters, {id: tableState.filterId});
```

Comment:
I think if the state.filters is undefined, we can use the `store.getState().filters.findWhere((filter) => filters.id === tableState.id + 'filter').filterText`

Reply:
I don't really understand.. It's allright if it's undefined the filter won't show up in the table if desired.

Reply:
I don't think we need to change the filter here, state will always be defined